test/services/LibraryService.test.ts

```diff
@@ -86,6 +147,178 @@ describe('LibraryService', () => {
+          expect(response.body.entry).toHaveLength(1);
+          expect(response.body.entry).toEqual(expect.arrayContaining([{ resource: LIBRARY_WITH_NO_DEPS }]));
+        });
+    });
+
+    it('returns a Bundle including the Library and its dependent libraries when the Library has dependencies and id passed through args', async () => {
+      await supertest(server.app)
+        .get('/4_0_1/Library/testLibraryWithDeps/$package')
+        .expect(200)
+        .expect(200)
+        .then(response => {
+          expect(response.body.resourceType).toEqual('Bundle');
+          expect(response.body.entry).toHaveLength(3);
+          expect(response.body.entry).toEqual(
+            expect.arrayContaining([
+              { resource: LIBRARY_WITH_NESTED_DEPS },
+              { resource: LIBRARY_WITH_NO_DEPS },
+              { resource: LIBRARY_WITH_DEPS }
+            ])
+          );
+        });
+    });
+
+    it('returns a Bundle including the Library and its dependent libraries when the Library has dependencies and id passed through body', async () => {
+      await supertest(server.app)
+        .post('/4_0_1/Library/$package')
+        .send({ resourceType: 'Parameters', parameter: [{ name: 'id', valueString: 'testLibraryWithDeps' }] })
+        .set('content-type', 'application/fhir+json')
+        .expect(200)
+        .then(response => {
+          expect(response.body.resourceType).toEqual('Bundle');
+          expect(response.body.entry).toHaveLength(3);
+          expect(response.body.entry).toEqual(
+            expect.arrayContaining([
+              { resource: LIBRARY_WITH_NESTED_DEPS },
+              { resource: LIBRARY_WITH_NO_DEPS },
+              { resource: LIBRARY_WITH_DEPS }
+            ])
+          );
+        });
+    });
+
+    it('returns a Bundle including just the Library when the Library has no dependencies and identifier with just idenifier.value passed through body', async () => {
+      await supertest(server.app)
+        .post('/4_0_1/Library/$package')
+        .send({
+          resourceType: 'Parameters',
+          parameter: [{ name: 'identifier', valueString: 'libraryWithIdentifierValue' }]
+        })
+        .set('content-type', 'application/fhir+json')
+        .expect(200)
+        .then(response => {
+          expect(response.body.resourceType).toEqual('Bundle');
+          expect(response.body.entry).toHaveLength(1);
+          expect(response.body.entry).toEqual(expect.arrayContaining([{ resource: LIBRARY_WITH_IDENTIFIER_VALUE }]));
+        });
+    });
+
+    it('returns a Bundle including just the Library when the Library has no dependencies and identifier with just identifier.system passed through body', async () => {
+      await supertest(server.app)
+        .post('/4_0_1/Library/$package')
+        .send({
+          resourceType: 'Parameters',
+          parameter: [{ name: 'identifier', valueString: 'http://example.com/libraryWithIdentifierSystem|' }]
```

Comment:
What if there are multiple Libraries with the same identifier system but different values?  Unless you are going to package them all, probably should expect 400.  IG should be clearer about how to handle such cases.

Reply:
I just added a test for this in my latest [commit](https://github.com/projecttacoma/measure-repository-service/pull/6/commits/b4d1898f62605f05632ecc2eace28683e5e56765). I hope this is what you had in mind!